prometheus-stats

```diff
@@ -67,14 +67,39 @@ def main():
             FROM TestRuns
             WHERE time > ? AND wait_seconds <= 3600; """, (since, )).fetchone()[0]
 
-    output += """# HELP queue_time_wait_seconds histogram of queue wait times in the last %i hours
+    output += f"""# HELP queue_time_wait_seconds histogram of queue wait times in the last {opts.hours} hours
 # TYPE queue_time_wait_seconds histogram
-queue_time_wait_seconds_bucket{le="300"} %i
-queue_time_wait_seconds_bucket{le="3600"} %i
-queue_time_wait_seconds_bucket{le="+Inf"} %i
-queue_time_wait_seconds_sum %i
-queue_time_wait_seconds_count %i
-""" % (opts.hours, test_runs_wait_5m, test_runs_wait_1h, test_runs, test_runs_queue_wait_sum, test_runs)
+queue_time_wait_seconds_bucket{{le="300"}} {test_runs_wait_5m}
+queue_time_wait_seconds_bucket{{le="3600"}} {test_runs_wait_1h}
+queue_time_wait_seconds_bucket{{le="+Inf"}} {test_runs}
+queue_time_wait_seconds_sum {test_runs_queue_wait_sum}
+queue_time_wait_seconds_count {test_runs}
+
+"""
+
+    # failures
+    top_failures = cursor.execute("""
+            SELECT TestRuns.project, t1.testname, (t2.failed * 100.0) / COUNT(run) AS percent
+            FROM Tests AS t1
+            JOIN TestRuns ON t1.run = TestRuns.id
+            JOIN (
+                SELECT testname, COUNT(run) as failed
+                FROM Tests
+                JOIN TestRuns ON Tests.run = TestRuns.id
+                WHERE TestRuns.time > strftime('%s', 'now', '-7 days') AND Tests.failed = 1
+                GROUP BY testname
+            ) AS t2 ON t1.testname = t2.testname
+            WHERE TestRuns.time > strftime('%s', 'now', '-7 days') AND t2.failed > 1
+            GROUP BY t1.testname
+            ORDER BY percent DESC
+            LIMIT 50""").fetchall()
+
+    output += """# HELP top_failures_pct Most unstable tests in the last 7 days (more than 10% failure rate)
```

Comment:
I am wondering if we are not doing this wrongly as one can query in graphana stats for lets say 24 hours and we will show "Most unstable tests in last 7 days from point of last 24 hours". (or we will show for every hour how it looks 7 days back?)

Reply:
I would not represent these as a graph, but as a table. It may be possible to get a selector for this like "how did that table look 3 days ago", but  at least initially the current list seems fine (what we also have on the weather report).

But admittedly I'm not at all versed in Grafana yet, so maybe there is some impedance mismatch here.

Reply:
ack, lets go with this then and see how it looks in graphana and how it works. If we realize we need something else, we can just change it.